### PR TITLE
Proxy scaffolding

### PR DESCRIPTION
**Pull request by samasno**
## Trello
https://trello.com/c/aj45KkKq/12-refactor-proxy-to-forward-requests-and-split-prod-dev-environments

## Setup
clone down repo
run npm install
run npm start
in browser navigate to `http://localhost:3000/description/1` for basic test

## Purpose of PR
build basic scaffolding for proxy server

### CODE REVIEW

File: controllers/errorHandling.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// return response in case of error
+module.exports = (err, req, res, next) => {
+  res.status(err.statis || 500).json(err.message || 'Server error');
```
Comment (tyler-thee-creator) [inline controllers/errorHandling.js:3]: Maybe I'm unaware of this syntax, but shouldn't this "read err.status" as opposed to "error.statis"? Either way, super cool that you have a file for error handling

File: routers/checkout.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const router = require('express').Router();
+router.route('....')
```
Comment (tyler-thee-creator) [inline routers/checkout.js:2]: This is cool, I've never seen this before

File: routers/description.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const { forwardingController } = require('../controllers/forwarding.js');
```
Comment (tyler-thee-creator) [inline routers/description.js:1]: Nice destructuring, super clean

File: server/index.js [hunk 4]
```diff
@@ -5,11 +5,13 @@ const port = 3000;
 const app = express();
 
 // For dev purposes
-// const morgan = require('morgan');
```
Comment (tyler-thee-creator) [inline server/index.js:8]: I've seen morgan and dev used in my code as well, not sure exactly what they do but know its related to development. I'll have to do research to see what their purpose is

**Review by tyler-thee-creator — APPROVED** [review]
Overall the code looks amazing. I think there may be 1 syntax error I pointed out but I could be wrong. Overall, super nice job, this code is fantastic

File: controllers/errorHandling.js (hunk 1 above)
Comment (samasno) [reply controllers/errorHandling.js:3]: embarassing typo. yeah that will return undefined and end up being 500 all the time

**Pull request merged**